reconcile/ocm_upgrade_scheduler.py

```diff
@@ -251,29 +269,18 @@ def calculate_diff(current_state, desired_state, ocm_map, version_history):
             continue
 
         # choose version that meets the conditions and add it to the diffs
-        available_upgrades = ocm.get_available_upgrades(
-            d["current_version"], d["channel"]
-        )
-        for version in reversed(sort_versions(available_upgrades)):
-            logging.debug(f"[{cluster}] checking conditions for version {version}")
-            if ocm.version_blocked(version):
-                logging.debug(f"[{cluster}] version {version} is blocked")
-                continue
-            if version_conditions_met(
-                version, version_history, ocm.name, d["workloads"], d["conditions"]
-            ):
-                logging.debug(f"[{cluster}] conditions met for version {version}")
-                item = {
-                    "action": "create",
-                    "cluster": cluster,
-                    "version": version,
-                    "schedule_type": "manual",
-                    "next_run": next_schedule.strftime("%Y-%m-%dT%H:%M:%SZ"),
-                }
-                for mutex in cluster_mutexes(d):
-                    locked[mutex] = cluster
-                diffs.append(item)
-                break
+        version = upgradeable_version(d, version_history, ocm)
+        if version:
+            item = {
+                "action": "create",
+                "cluster": cluster,
+                "version": version,
+                "schedule_type": "manual",
+                "next_run": next_schedule.strftime("%Y-%m-%dT%H:%M:%SZ"),
+            }
+            for mutex in cluster_mutexes(d):
+                locked[mutex] = cluster
```

Comment:
Out of curiosity: Doesnt that mean that any already held mutex for that cluster gets overridden? The logic is consistent with the current master state, however, it kinda puzzles me :sweat_smile: Would have thought that if the mutex alrdy exists, there will be a skip.

Reply:
This locks all mutexes for that cluster. The check if the mutexes are locked is at line 264